### PR TITLE
feat(payment): PI-1606 [Afterpay] Move Afterpay payment strategy to separate package in checkout-sdk-js

### PR DESCRIPTION
**Pull request by bc-wpietrzak**
## What?
Move Afterpay to separate package in checkout-sdk-js.

## Why?
For better separation between payment integrations and core Checkout SDK.

## Testing / Proof
Manual + unit testing

Screenshot 2024-04-05 at 02 17 49

https://github.com/bigcommerce/checkout-sdk-js/assets/138816572/eb9ede7f-97f7-4c66-8d1c-bdda67e57b94

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/afterpay-integration/src/afterpay-payment-strategy.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        await this._loadPaymentMethod(gatewayId, methodId, options);
+
+        state = this._paymentIntegrationService.getState();
+
+        this._redirectToAfterpay(countryCode, state.getPaymentMethod(methodId, gatewayId));
+
+        // Afterpay will handle the rest of the flow so return a promise that doesn't really resolve
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        return new Promise<never>(() => {});
```
[the remark this answers is not in the document]
Comment (bc-wpietrzak) [reply packages/afterpay-integration/src/afterpay-payment-strategy.ts:80]: Without `return new Promise<never>(() => {});`, when we place order we get redirected directly to order confirmation screen, skipping redirect to afterpay completely

File: packages/afterpay-integration/src/afterpay-payment-strategy.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+        await this._paymentIntegrationService.submitOrder({}, options);
+
+        try {
+            await this._paymentIntegrationService.submitPayment(paymentPayload);
+        } catch (error) {
+            await this._paymentIntegrationService.forgetCheckout(payment.providerId);
+            await this._paymentIntegrationService.loadPaymentMethods();
+
+            throw new OrderFinalizationNotCompletedError();
```
[the remark this answers is not in the document]
Comment (bc-wpietrzak) [reply packages/afterpay-integration/src/afterpay-payment-strategy.ts]: Fixed

File: packages/afterpay-integration/src/afterpay-payment-strategy.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+        if (useStoreCredit !== undefined) {
+            await this._paymentIntegrationService.applyStoreCredit(useStoreCredit);
+        }
+
+        await this._loadPaymentMethod(gatewayId, methodId, options);
+
+        state = this._paymentIntegrationService.getState();
+
+        this._redirectToAfterpay(countryCode, state.getPaymentMethod(methodId, gatewayId));
```
[the remark this answers is not in the document]
Comment (bc-wpietrzak) [reply packages/afterpay-integration/src/afterpay-payment-strategy.ts:78]: Fixed

File: packages/core/src/payment-integration/default-payment-integration-service.ts [hunk 4]
```diff
@@ -110,6 +110,12 @@ export default class DefaultPaymentIntegrationService implements PaymentIntegrat
         return this._storeProjection.getState();
     }
 
+    async loadPaymentMethods(options?: RequestOptions): Promise<PaymentIntegrationSelectors> {
```
[the remark this answers is not in the document]
Comment (bc-wpietrzak) [reply packages/core/src/payment-integration/default-payment-integration-service.ts]: Moved those changes to a separate PR (already merged)

**Review by animesh1987 — APPROVED** [review]

**Review by vitalii-koshovyi — APPROVED** [review]

**Pull request merged**